azurerm/internal/services/media/tests/media_asset_resource_test.go

```diff
@@ -0,0 +1,212 @@
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMMediaAsset_custom_container(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_media_asset", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMMediaAssetDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMMediaAsset_custom_container(data),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(data.ResourceName, "container", "assetcontainer"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMMediaAsset_storage_account(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_media_asset", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMMediaAssetDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMMediaAsset_storage_account(data),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(data.ResourceName, "storage_account_name", fmt.Sprintf("acctestsa1%s", data.RandomString)),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func testCheckAzureRMMediaAssetDestroy(s *terraform.State) error {
+	conn := acceptance.AzureProvider.Meta().(*clients.Client).Media.AssetsClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_media_asset" {
+			continue
+		}
+
+		id, err := parse.MediaAssetsID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+		resp, err := conn.Get(ctx, id.ResourceGroup, id.AccountName, id.Name)
+
+		if err != nil {
+			return nil
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("Media Asset still exists:\n%#v", resp)
+		}
+	}
+
+	return nil
+}
+
+func testAccAzureRMMediaAsset_basic(data acceptance.TestData) string {
+	template := testAccAzureRMMediaAsset_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_media_asset" "test" {
+  name                        = "asset%s"
+  description                 = "Asset description"
+  resource_group_name         = azurerm_resource_group.test.name
+  media_services_account_name = azurerm_media_services_account.test.name
+}
+
+`, template, data.RandomString)
+}
+
+func testAccAzureRMMediaAsset_alternate_id(data acceptance.TestData) string {
+	template := testAccAzureRMMediaAsset_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_media_asset" "test" {
+  name                        = "asset%s"
+  description                 = "Asset description"
+  resource_group_name         = azurerm_resource_group.test.name
+  media_services_account_name = azurerm_media_services_account.test.name
+  alternate_id                = "assetalternateid"
```

Comment:
same here, we sould always have some capitals if possible to ensure the API handles them